parcels/grid.py

```diff
@@ -87,20 +88,91 @@ def child_ctypes_struct(self):
         """Returns a ctypes struct object containing all relevant
         pointers and sizes for this grid."""
 
-        class CStructuredGrid(Structure):
+        class CRectilinearGrid(Structure):
+            # z4d is only to have same cstruct as RectilinearSGrid
             _fields_ = [('xdim', c_int), ('ydim', c_int), ('zdim', c_int),
-                        ('tdim', c_int), ('tidx', c_int),
+                        ('tdim', c_int), ('tidx', c_int), ('z4d', c_int),
                         ('lon', POINTER(c_float)), ('lat', POINTER(c_float)),
                         ('depth', POINTER(c_float)), ('time', POINTER(c_double))
                         ]
 
         # Create and populate the c-struct object
-        cstruct = CStructuredGrid(self.lon.size, self.lat.size, self.depth.size,
-                                  self.time.size, 0,
-                                  self.lon.ctypes.data_as(POINTER(c_float)),
-                                  self.lat.ctypes.data_as(POINTER(c_float)),
-                                  self.depth.ctypes.data_as(POINTER(c_float)),
-                                  self.time.ctypes.data_as(POINTER(c_double)))
+        cstruct = CRectilinearGrid(self.lon.size, self.lat.size, self.depth.size,
+                                   self.time.size, 0, -1,
+                                   self.lon.ctypes.data_as(POINTER(c_float)),
+                                   self.lat.ctypes.data_as(POINTER(c_float)),
+                                   self.depth.ctypes.data_as(POINTER(c_float)),
+                                   self.time.ctypes.data_as(POINTER(c_double)))
+        return cstruct
+
+
+class RectilinearSGrid(Grid):
+    """Rectilinear S Grid. Same horizontal discretisation as a rectilinear grid,
+       but with s vertical coordinates
```

Comment:
Be more explicit here that with `s vertical coordinates`, we mean terrain-following (sigma) or density (rho) layers